src/utils/src/types/work-package-types.ts

```diff
@@ -4,112 +4,39 @@
-    userId: { type: 'integer', minimum: 0 },
-    name: { type: 'string' },
-    crId: { type: 'integer', minimum: 0 },
-    projectId: { type: 'integer', minimum: 0 },
-    startDate: { type: 'string', format: 'date' },
-    duration: { type: 'integer', minimum: 0 },
-    wbsElementIds: {
-      type: 'array',
-      items: {
-        type: 'integer',
-        minimum: 0
-      }
-    },
-    expectedActivities: {
-      type: 'array',
-      items: {
-        type: 'string',
-        minLength: 0
-      }
-    },
-    deliverables: {
-      type: 'array',
-      items: {
-        type: 'string',
-        minLength: 0
-      }
-    }
-  },
-  required: [
-    'userId',
-    'name',
-    'crId',
-    'projectId',
-    'startDate',
-    'duration',
-    'wbsElementIds',
-    'expectedActivities',
-    'deliverables'
-  ]
-} as const;
+export const workPackageCreateInputSchemaBody = bodySchema({
+  userId: intType,
+  name: stringType,
+  crId: intType,
+  projectId: intType,
+  startDate: dateType,
+  duration: intType,
+  dependencies: arrayType(intType),
+  expectedActivities: arrayType(stringType),
+  deliverables: arrayType(stringType)
+});
 
 export type CreateWorkPackagePayload = FromSchema<typeof workPackageCreateInputSchemaBody>;
 
-export const workPackageEditInputSchemaBody = {
-  type: 'object',
-  properties: {
-    wbsElementId: { type: 'integer', minimum: 0 },
-    userId: { type: 'integer', minimum: 0 },
-    name: { type: 'string' },
-    crId: { type: 'integer', minimum: 0 },
-    startDate: { type: 'string', format: 'date' },
-    duration: { type: 'integer', minimum: 0 },
-    wbsElementIds: {
-      type: 'array',
-      items: {
-        type: 'integer',
-        minimum: 0
-      }
-    },
-    expectedActivities: {
-      type: 'array',
-      items: {
-        type: 'object',
-        properties: {
-          id: { type: 'integer', minimum: 0 },
-          detail: { type: 'string' }
-        },
-        required: ['id', 'detail']
-      }
-    },
-    deliverables: {
-      type: 'array',
-      items: {
-        type: 'object',
-        properties: {
-          id: { type: 'integer', minimum: 0 },
-          detail: { type: 'string' }
-        },
-        required: ['id', 'detail']
-      }
-    },
-    wbsElementStatus: {
-      type: 'string',
-      enum: ['INACTIVE', 'ACTIVE', 'COMPLETE']
-    },
-    progress: { type: 'integer', minimum: 0 },
-    projectLead: { type: 'integer', minimum: 0 },
-    projectManager: { type: 'integer', minimum: 0 }
+export const workPackageEditInputSchemaBody = bodySchema(
+  {
+    wbsElementId: intType,
+    userId: intType,
+    name: stringType,
+    crId: intType,
+    startDate: dateType,
+    duration: intType,
+    dependencies: arrayType(intType),
+    expectedActivities: arrayType(bodySchema({ id: intType, detail: stringType })),
+    deliverables: arrayType(bodySchema({ id: intType, detail: stringType })),
+    wbsElementStatus: enumType('INACTIVE', 'ACTIVE', 'COMPLETE'),
```

Comment:
can this use the proper enum instead of hard-coded strings?

Reply:
Yup, if the enum is turned into a union type + map of key values, it could also use just ...Object.values(), but that's for another time